# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Lowering.cpp
# Commit: 69a27bde5003
# Full Hash: 69a27bde5003c70c97e8c31202c1fcf3e13d173e
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 09:30:56
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.4: Add tls dependency for WasmTruncateToInt64 and Int64ToFloatingPoint for arm. r=lth
#   
#   In this patch we add a tls dependency for the remaining nodes which use
#   BuiltinThunk to call c++ runtime. By ABI requirements WasmTlsReg should
#   be set.
# ==============================================================================

diff -r 9b42e0b690b6 -r 69a27bde5003 js/src/jit/Lowering.cpp
--- a/js/src/jit/Lowering.cpp	Wed Sep 09 09:50:35 2020 +0000
+++ b/js/src/jit/Lowering.cpp	Wed Sep 09 10:48:09 2020 +0000
@@ -1853,6 +1853,16 @@
   lowerWasmBuiltinModI64(mod);
 }
 
+void LIRGenerator::visitBuiltinInt64ToFloatingPoint(
+    MBuiltinInt64ToFloatingPoint* ins) {
+  lowerBuiltinInt64ToFloatingPoint(ins);
+}
+
+void LIRGenerator::visitWasmBuiltinTruncateToInt64(
+    MWasmBuiltinTruncateToInt64* ins) {
+  lowerWasmBuiltinTruncateToInt64(ins);
+}
+
 void LIRGenerator::visitMod(MMod* ins) {
   MOZ_ASSERT(ins->lhs()->type() == ins->rhs()->type());
   MOZ_ASSERT(IsNumberType(ins->type()));